### ISSUE TITLE
Cannot read properties of undefined (reading 'startConnecting')

### ISSUE THREAD

**Issue by akosyakov** [grpc/grpc-node#2537]
After upgrading 1.8.17 we see lot of

```
TypeError: Cannot read properties of undefined (reading 'startConnecting')
    at /app/node_modules/@grpc/grpc-js/src/load-balancer-pick-first.ts:308:43
    at processTicksAndRejections (node:internal/process/task_queues:78:11)
```

We tried to downgrade a bit to 1.8.8 but it still happens from time to time.

**Comment by murgatroid99** [comment]
Can you check if the same thing happens with version 1.9.0?

**Comment by svenefftinge** [comment]
I believe it can potentially still happen as [the code](https://github.com/grpc/grpc-node/blob/master/packages/grpc-js/src/load-balancer-pick-first.ts#L309) is executed on next tick and there's a on of other code pathes potentially being executed that would potentially change/clear the `children` property. 
So I'd say either capturing the `children` array in the lambda or checking for `undefined` would be important.

**Comment by murgatroid99** [comment]
The fix for this has been published in version 1.9.1.
